stunnel/stunnel.ml

```diff
@@ -140,35 +140,40 @@ let config_file verify_cert extended_diagnosis host port legacy =
     | Some s -> s
   in
 
-  let lines = [
-    "client=yes"; "foreground=yes"; "socket = r:TCP_NODELAY=1"; "socket = r:SO_KEEPALIVE=1"; "socket = a:SO_KEEPALIVE=1";
-    (match !timeoutidle with None -> "" | Some x -> Printf.sprintf "TIMEOUTidle = %d" x);
-    Printf.sprintf "connect=%s:%d" host port;
-    "fips = no"; (* stunnel fips-mode stops us using sslVersion other than TLSv1 which means 1.0 only. *)
-  ] @	(if extended_diagnosis then
-         ["debug=4"]
-       else
-         []
-      ) @ (
-      if verify_cert then
-        ["verify=2";
-         sprintf "CApath=%s" certificate_path;
-         sprintf "CRLpath=%s" crl_path]
-      else
-        []
-    ) @ (
-      if legacy then [
-        "sslVersion = all";
-        "options = NO_SSLv2";
-        "options = NO_SSLv3";
-        "ciphers = " ^ (good_ciphers ()) ^ (match !legacy_ciphersuites with "" -> "" | s -> (":" ^ s))
-      ] else [
-        "sslVersion = TLSv1.2";
-        "ciphers = " ^ (good_ciphers ());
+  let lines = 
+    String.concat "\n" @@ List.concat
+    [ [ "client=yes"
+      ; "foreground=yes"
+      ; "socket = r:TCP_NODELAY=1"
+      ; "socket = r:SO_KEEPALIVE=1"
+      ; "socket = a:SO_KEEPALIVE=1"
+      ; (match !timeoutidle with
+        None -> "" |
+        Some x -> Printf.sprintf "TIMEOUTidle = %d" x)
+      ; Printf.sprintf "connect=%s:%d" host port
+      ; "fips = no"; (* stunnel fips-mode stops us using sslVersion other than TLSv1 which means 1.0 only. *)
       ]
-    )
+    ; if extended_diagnosis then ["debug=4"] else []
+    ; if verify_cert then
+        ["verify=2"
+        ; sprintf "CApath=%s" certificate_path
+        ; sprintf "CRLpath=%s" crl_path
+        ]
+      else []
+    ; if legacy then
+        [ "sslVersion = all"
+        ; "options = NO_SSLv2"
+        ; "options = NO_SSLv3"
+        ; "ciphers = " ^ (good_ciphers ()) ^ (match !legacy_ciphersuites with "" -> "" | s -> (":" ^ s))
+        ]
+      else 
+        [ "sslVersion = TLSv1.2"
+        ; "ciphers = " ^ (good_ciphers ())
+        ]
+      ; ["curve = secp384r1"]
+    ]
```

Comment:
To make sure the last line has a newline, you could add one empty string as the last element. String.concat only inserts newlines between lines.

```
String.concat "\n" @@ List.concat
[ ...
; [""] (* ensure \n at last line above *)
]
```

Reply:
Done.